### PR TITLE
Features/profile picture1

### PR DESCRIPTION
**Pull request by pokumars**
1. When a user registers, they get an auto-generated username.
2. User can change the username in the profile page.

### CODE REVIEW

**Comment by stahamnguyen** [comment]
There are conflicts in `components/Profile/ProfileUserDetail.js` and `constants/color.js` as Github said. Please solve the conflict locally and push the update to this PR @pokumars

**Comment by stahamnguyen** [comment]
Also, try to pull the remote `master` branch to this branch and resolves the conflicts

File: components/Profile/DetailUpdateModal.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    presentationStyle="fullScreen"
+    onRequestClose={() => {
+      toggleDetailModal();
+      Alert.alert('Username change aborted');
+    }}
+  >
+    <View style={styles.container}>
+      <Text style={styles.errorMessage} >{errorMessage}</Text>
+      <TextInput style={styles.input}
+        onChangeText={(text) => setDetail(text)}
```
Comment (stahamnguyen) [inline components/Profile/DetailUpdateModal.js]: This could be simply `onChangeText={setDetail}`

File: components/Profile/DetailUpdateModal.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+const DetailUpdateModal = ({toggleDetailModal, modalVisible, originalDetail, onConfirm}) => {
+  const [detail, setDetail] = React.useState('');
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const legalChars = /^[a-zA-Z0-9_-]*$/; // allow letters, numbers, and underscores
+
+  const updateUserDetails = () => {
+    if (detail.length < 5){
+      setErrorMessage('The username must be longer than 5 characters');
+      return;
+    } else if (!legalChars.test(detail)){
```
Comment (stahamnguyen) [inline components/Profile/DetailUpdateModal.js]: No need of `else` here and below. Just `if` is enough

File: screens/ProfileScreen.js [hunk 3]
```diff
@@ -32,7 +36,18 @@ const ProfileScreen = ({navigation}) => {
       );
     console.log('Sign_out clicked');
   };
-  /* 
+
+  const updateUsername = (newUsername) => {
+    console.log('ProfileScreen received this', newUsername);
+    auth()
```
Comment (stahamnguyen) [inline screens/ProfileScreen.js]: Should `return` here (return a `Promise`) to make the `updateUsername` chainable, waiting for it to be successful to run `setDetail` and `toggleDetailModal` in `DetailUpdateModal`

File: components/Profile/DetailUpdateModal.js (hunk 1 above)
Comment (pokumars) [reply components/Profile/DetailUpdateModal.js]: I see.
Isn't that just a short form of writing the same thing? Or what is the difference?
Is it just for the appearance or there is some difference? performance benefits

Comment (stahamnguyen) [reply components/Profile/DetailUpdateModal.js]: It's both a shorter form and more performant. `(text) => setDetail(text)` is a new function every time the `onChangeText` gets called, while `setDetail` will be referenced if using the shorter form (no new function initialization)

File: components/Profile/DetailUpdateModal.js (hunk 2 above)
Comment (pokumars) [reply components/Profile/DetailUpdateModal.js]: elses removed

File: screens/ProfileScreen.js [hunk 4]
```diff
@@ -44,7 +66,7 @@ const ProfileScreen = ({ navigation }) => {
             textFirst
             btnText="Logout"
             btnImage={require('../assets/icons/log-out.png')}
-            onPress={() => Sign_out()}
+            onPress={() => signOut()}
```
Comment (stahamnguyen) [inline screens/ProfileScreen.js]: `onPress={signOut}` is enough

**Review by stahamnguyen — APPROVED** [review]

**Pull request merged**